### PR TITLE
Rebase minio/parquet-go and fix null handling.

### PR DESCRIPTION
**Pull request by balamurugana**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Comment by harshavardhana** [comment]
@balamurugana this PR crashes with our `testdata.parquet`

```
2019-01-11 01:53:59.551198 I | http: panic serving 127.0.0.1:43114: runtime error: slice bounds out of range
goroutine 92 [running]:
net/http.(*conn).serve.func1(0xc000329900)
	/home/harsha/.gimme/versions/go1.11.4.linux.amd64/src/net/http/server.go:1746 +0xd0
panic(0x140a480, 0x2683fe0)
	/home/harsha/.gimme/versions/go1.11.4.linux.amd64/src/runtime/panic.go:513 +0x1b9
github.com/minio/minio/cmd.criticalErrorHandler.ServeHTTP.func1(0x17ec960, 0xc0004427e0, 0xc00016ea00)
	/home/harsha/mygo/src/github.com/minio/minio/cmd/generic-handlers.go:812 +0xaa
panic(0x140a480, 0x2683fe0)
	/home/harsha/.gimme/versions/go1.11.4.linux.amd64/src/runtime/panic.go:513 +0x1b9
github.com/minio/minio/vendor/github.com/minio/parquet-go.readDataPageValues(0xc0005e43a8, 0x2, 0x5, 0xffffffffffffffff, 0x2, 0x0, 0x135c640, 0xc0004ec160, 0x2, 0x0, ...)
	/home/harsha/mygo/src/github.com/minio/minio/vendor/github.com/minio/parquet-go/decode.go:455 +0x943
github.com/minio/minio/vendor/github.com/minio/parquet-go.readPage(0xc00002bde0, 0xc0002f5550, 0xc000489da0, 0xc000489d70, 0x5, 0x5, 0x41f898, 0xc000039710, 0xc0002d1cb0, 0xc00031e708, ...)
	/home/harsha/mygo/src/github.com/minio/minio/vendor/github.com/minio/parquet-go/page.go:262 +0x5f4
github.com/minio/minio/vendor/github.com/minio/parquet-go.(*column).readPage(0xc0000c1100)
	/home/harsha/mygo/src/github.com/minio/minio/vendor/github.com/minio/parquet-go/column.go:110 +0x71
github.com/minio/minio/vendor/github.com/minio/parquet-go.(*column).readPage(0xc0000c1100)
	/home/harsha/mygo/src/github.com/minio/minio/vendor/github.com/minio/parquet-go/column.go:124 +0xa9
github.com/minio/minio/vendor/github.com/minio/parquet-go.(*column).read(0xc0000c1100, 0xc000272c60, 0xc00046a2b0, 0x3)
	/home/harsha/mygo/src/github.com/minio/minio/vendor/github.com/minio/parquet-go/column.go:139 +0xdc
github.com/minio/minio/vendor/github.com/minio/parquet-go.(*File).Read(0xc0002e50e0, 0x2, 0x1, 0xc00065e000)
	/home/harsha/mygo/src/github.com/minio/minio/vendor/github.com/minio/parquet-go/parquet.go:139 +0x141
github.com/minio/minio/pkg/s3select/parquet.(*Reader).Read(0xc000385850, 0xc0002f0600, 0x0, 0xc00056af00, 0xc000385850)
	/home/harsha/mygo/src/github.com/minio/minio/pkg/s3select/parquet/reader.go:36 +0x47
github.com/minio/minio/pkg/s3select.(*S3Select).Evaluate(0xc0002e2340, 0x17e7260, 0xc0002f0600)
	/home/harsha/mygo/src/github.com/minio/minio/pkg/s3select/select.go:337 +0x14e
```

```
~ cp ../minio/pkg/s3select/testdata.parquet myminio/sjm-airlines/
```
```
~ mc sql -e "select * from s3object" myminio/sjm-airlines/testdata.parquet
```

Using this PR for `mc` https://github.com/minio/mc/pull/2643

**Comment by harshavardhana** [comment]
Also downloaded https://raw.githubusercontent.com/Teradata/kylo/master/samples/sample-data/parquet/userdata1.parquet returned value is empty

```
~ mc sql -e "select * from s3object" myminio/sjm-airlines/userdata1.parquet | head -1
{"birthdate":null,"cc":null,"last_name":null,"id":null,"registration_dttm":"\u0000*\ufffdl\ufffd\u0019\u0000\u0000N%\u0000","comments":null,"email":"ajordan0@com.com","first_name":null,"title":null,"country":null,"salary":null,"ip_address":null,"gender":null}
```

With S3
```
~ mc sql -e "select * from s3object" s3/sjm-airlines-1/userdata1.parquet | head -1
{"registration_dttm":45331434715132002751293952,"id":1,"first_name":"Amanda","last_name":"Jordan","email":"ajordan0@com.com","gender":"Female","ip_address":"1.197.201.2","cc":"6759521864920116","country":"Indonesia","birthdate":"3/8/1971","salary":49756.53e0,"title":"Internal Auditor","comments":"1E+02"}
```

**Comment by balamurugana** [comment]
@harshavardhana The crash and returning null are bugs in minio/parquet-go.  They need to be opened in minio/parquet-go.  IMO those issues are not related to this PR.

**Comment by harshavardhana** [comment]
> @harshavardhana The crash and returning null are bugs in minio/parquet-go. They need to be opened in minio/parquet-go. IMO those issues are not related to this PR.

@balamurugana my point is it was working before, I can't even use the testdata.parquet for Minio's implementation. Because I don't know what this PR fixes, the intention was to test this PR.

**Comment by balamurugana** [comment]
@harshavardhana No. it didn't.  See https://github.com/minio/parquet-go/pull/7

**Comment by harshavardhana** [comment]
> @harshavardhana No. it didn't. See [minio/parquet-go#7](https://github.com/minio/parquet-go/pull/7)

I tried all the three parquet files they seem to be working fine, now with a recent update on the PR.

**Comment by harshavardhana** [comment]
```
~  mc sql -e "select * from s3object" myminio/sjm-airlines/people.parquet
{"name":"Michael","age":31}
{"name":"Andy","age":30}
{"name":"Justin","age":19}
```

```
~ mc sql -e "select * from s3object" myminio/sjm-airlines/people.parquet
{"age":31,"name":"Michael"}
{"age":30,"name":"Andy"}
{"age":19,"name":"Justin"}
```

```
~ mc sql -e "select * from s3object" myminio/sjm-airlines/people.parquet
{"age":31,"name":"Michael"}
{"name":"Andy","age":30}
{"age":19,"name":"Justin"}
```

I am not sure why we change the output in this manner, AWS S3 makes sure that the returned output is consistent. There is some incoming order issue in parquetgo which leads to this behavior basically. 

Where name and age arrive differently, this is perhaps due to the `map[string]Value` usage in parquetgo.File.Read() for both returned value and also column names. To fix this I ended up using slices which are better at giving consistent outputs @balamurugana 

Will send a PR to parquet-go.

**Comment by harshavardhana** [comment]
@kannappanr @nitisht we need this PR to go in for the release as well, we have prospective users waiting and current master or release - I cannot use parquet with select with different types of parquet files.

@sinhaashish @donatello ping

File: pkg/s3select/sql/value.go [hunk 1]
```diff
@@ -66,6 +70,15 @@ func (value *Value) MarshalJSON() ([]byte, error) {
 	return json.Marshal(value.value)
 }
 
+// NullValue - returns underlying null value. It panics if value is not null type.
+func (value *Value) NullValue() *struct{} {
+	if value.valueType == Null {
+		return nil
+	}
+
+	panic(fmt.Sprintf("requested bool value but found %T type", value.value))
```
Comment (kannappanr) [inline pkg/s3select/sql/value.go:79]: Is there  typo, should this be
`panic(fmt.Sprintf("requested null value but found %T type", value.value))`

**Review by harshavardhana — APPROVED** [review]
LGTM tested

**Review by donatello — APPROVED** [review]

**Pull request merged**